### DISCUSSION TITLE
Inifite segLength

### DISCUSSION THREAD

**Discussion by niamaradi** [mtex-toolbox/mtex#1495] [Ask Anything]
Hello Mtex community, 

I want to calculate a LAB/HAB seglength ratio and I did this by calculating the sum of the segLength of both the LAB and HAB. However, with one particular ebsd map. I get an infinite segLength sum for the HAB, is there a way to avoid having infinite segLengths when computing the grains ? 

Btw this is how I compute grains 

`ebsd=ebsd.gridify;
[grains,ebsd.grainId] = calcGrains(ebsd,'angle',[15 2]*degree,'boundary','tight');
ebsd(grains(grains.grainSize<=4)) = [];
[grains,ebsd.grainId] = calcGrains(ebsd('Ni'),'angle',[15 2]*degree,'boundary','tight');
`

Thank you in advance for your help.
Niama

**Comment by kilir** [comment]
Hi,
while it might take some time with EBSD to map a grain structure with infinite segLength, it so sounds pretty much like a bug. Can you sort out the problematic grain or the region of the map and submit?

Finding the grain:
```
grains('id',grains.boundary.grainId(grains.boundary.segLength == inf))
```

Cheers,
Rüdiger

**Comment by niamaradi** [comment]
Hello Rüdiger, thank you for your quick reply, 

When I try finding the grains id with your method I get this error 

`Error using subsind (line 10)
No data with the specified ids in the data set

Error in grain2d/subsref (line 61)
    ind = subsind(grains,s(1).subs);`

However when I do the following 

`condinf=grains.boundary.segLength == inf
find(condinf==1)`
I get 

`ans =

     1
     2`

Now if I do `grains.boundary.grainId(1,2)` I get the grainId which is 352, that in my map is the "biggest grain" which is in reality my single crystal substrate. 

Is there any way I can still include it without having the infinite segLength ? 

Thank you,
Niama

**Comment by ralfHielscher** [comment]
Could you please try 
```matlab
[grains,ebsd.grainId] = calcGrains(ebsd,'angle',[15 2]*degree,'boundary','tight','noQHull');
```
Ralf.

**Reply by niamaradi** [reply]
Hello Ralf, 

I tried it and had no longer the segLength issue, thank you ! Can you please explain what does the 'noQHull' option add ? 

Thank you one more time for your help.

Kind regards,
Niama
